src/macrofier/engine.mjs

```diff
@@ -625,21 +625,36 @@ function generateSchemas(json, templates, options) {
     results.push(result)
   }
 
+  const list = []
+
   // schemas may be 1 or 2 levels deeps
   Object.entries(schemas).forEach( ([name, schema]) => {
     if (isSchema(schema)) {
-      generate(name, schema)
+      list.push([name, schema])
     }
     else if (typeof schema === 'object') {
       const uri = schema.uri
       Object.entries(schema).forEach( ([name, schema]) => {
         if (name !== 'uri') {
-          generate(name, schema, uri)
+          list.push([name, schema, uri])
         }
       })
     }
   })
 
+  list.sort((a, b) => {
+    const aInB = isDefinitionReferencedBySchema('#/components/schemas/' + a[0], b[1])
```

Comment:
You'll need to check for the presence of `uri` (third array item).

If it's present, then the path to the schema is `/x-schemas/${module.info.title}` and we didn't grab the title in the loop above, so we don't have it down here.

I recommend removing the else above, to keep this code simpler.